SentenceVectors.ecl

```diff
@@ -0,0 +1,641 @@
+    w := int.Weights(shape);  // Module to manage weights
+    // Extract the vectors for each word.  Words should be evenly distributed at this point.
+    WordInfo getVectors(WordInfo wi) := TRANSFORM
+      // The weights that form the word vector are the layer 1 weights.  j is the wordId
+      // and i is the term number of the word vector.  The terms of the word vector are
+      // contiguous, so we only need to know the start and end indexes.
+      startIndx := w.toFlatIndex(1, wi.wordId, 1);
+      endIndx := startIndx + shape[2] - 1;
+      // The weights might span slices.
+      startSlice := (startIndx - 1) DIV w.sliceSize + 1;
+      endSlice := (endIndx - 1) DIV w.sliceSize + 1;
+      // Get the start and end positions within the slice.
+      locStart := (startIndx-1) % w.sliceSize + 1;
+      locEnd := (endIndx - 1) % w.sliceSize + 1;
+      // If the start slice is the same as the end slice, extract the weights from
+      // that slice.  If different, concatenate the weights from the end of the start
+      // slice and the beginning of the end slice.
+      locSlices := slices(nodeId = node);
+      slice1 := locSlices(sliceId = startSlice)[1].weights[locStart .. locEnd];
+      slice2 := locSlices(sliceId = startSlice)[1].weights[locStart ..] + 
+                locSlices(sliceId = endSlice)[1].weights[.. locEnd];
+      vecWeights := IF(startSlice = endSlice, slice1, slice2);
+      // Normalize the weights to get the word vector
+      SELF.vec := Utils.normalizeVector(vecWeights);
+      SELF := wi;
+    END;
+    wordsWithVecs := PROJECT(words, getVectors(LEFT), LOCAL);
+    RETURN wordsWithVecs;
+  END;
+  /**
+    * Convert the vectorized vocabulary into the word portion of the model.
+    */
+  SHARED DATASET(TextMod) makeWordModel(DATASET(WordInfo) words) := FUNCTION
+    modOut := PROJECT(words, TRANSFORM(TextMod, SELF.typ := Types.t_ModRecType.Word,
+                          SELF.id := LEFT.wordId, SELF := LEFT), LOCAL);
+    RETURN modOut;
+  END;
+  /**
+    * Convert the set of vectorized sentences into the sentence portion of the model.
+    */
+  SHARED DATASET(TextMod) makeSentModel(DATASET(SentInfo) sent) := FUNCTION
+    modOut := PROJECT(sent, TRANSFORM(TextMod, SELF.typ := Types.t_ModRecType.Sentence,
+                          SELF.id := LEFT.sentId, SELF := LEFT), LOCAL);
+    RETURN modOut;
+  END;
+  /**
+    * Try to find the closest match between an input set of words and words that are in the
+    * vocabulary using Edit Distance and Numeric Distance.
+    */ 
+  SHARED DATASET(mappingRec) findClosestMatch(DATASET(TextMod) mod, DATASET(mappingRec) words) := FUNCTION
+    distRec := RECORD(mappingRec)
+      UNSIGNED4 eDist;
+    END;
+    wordsN := words(Utils.isNumeric(orig));
+    wordsT := words(Utils.isNumeric(orig)=FALSE);
+    d0T := JOIN(wordsT, mod(Utils.isNumeric(text)=FALSE), TRUE,
+                      TRANSFORM(distRec, SELF.eDist := Str.EditDistance(LEFT.orig, RIGHT.text),
```

Comment:
This would be significantly more efficient if you had a version of edit distance that had a maximum distance.  There is already a function that returns true/false for a given edit distance.  Please open a jira - it would be trivial to add.